lua/cmp/utils/async.lua

```diff
@@ -32,18 +32,24 @@ async.throttle = function(fn, timeout)
   local timer = assert(vim.loop.new_timer())
   local _async = nil ---@type Async?
   timers[#timers + 1] = timer
-  return setmetatable({
+  local throttle
+  throttle = setmetatable({
     running = false,
     timeout = timeout,
     sync = function(self, timeout_)
+      if not self.running then
+        return
+      end
       vim.wait(timeout_ or 1000, function()
         return not self.running
-      end)
+      end, 10)
```

Comment:
This is the real fix for the delay some users experienced.

Reason is that previsouly the code was not async, so filter would not be running in most cases.
With async, we are hitting the wait in `cmp.sync`.

Default interval is `200ms` for `vim.wait` which was causing the delays.

Reply:
thank you, i will check it now

Reply:
@hrsh7th instead of `10`, it could be set to an even lower value, but I personally don't see the delays anymore in the repro.

Reply:
It's still pretty bad, I tried this:
```lua
cmp.visible = cmp.sync(function()
  return cmp.core.view:visible() or vim.fn.pumvisible() == 1
end)
```
to
```lua
cmp.visible = function()
  return cmp.core.view:visible() or vim.fn.pumvisible() == 1
end
```

and the delay is gone. I have no idea about the architecture of `nvim-cmp` and why a visibility check would require so much code.

Reply:
@mgnsk the only repro that was provided to me is [here](https://github.com/hrsh7th/nvim-cmp/pull/1610#issuecomment-1584280889)

Those delays are fixed by this PR and multiple people confirmed it.

Are you 100% you're on the latest nvim-cmp version?

If the issue persists, then I really need a repro to check.

Reply:
I'll look into it more closely in the evening. I'm sure I saw this PR title when I did `Lazy sync`.